aiidalab_widgets_base/elns.py

```diff
@@ -0,0 +1,361 @@
+
+        self.eln_instance = ipw.Dropdown(
+            description="ELN:",
+            options=("Setup new ELN", {}),
+            style={"description_width": "initial"},
+        )
+        self.update_list_of_elns()
+
+        self.eln_types = ipw.Dropdown(
+            description="ELN type:",
+            options=["cheminfo", "openbis"],
+            value="cheminfo",
+            style={"description_width": "initial"},
+        )
+
+        self.eln_instance.observe(self.display_eln_config, names=["value", "options"])
+        self.eln_types.observe(self.display_eln_config, names=["value", "options"])
+
+        default_button = ipw.Button(description="Set as default", button_style="info")
+        default_button.on_click(self.set_current_eln_as_default)
+
+        save_config = ipw.Button(
+            description="Save configuration", button_style="success"
+        )
+        save_config.on_click(self.save_eln_configuration)
+
+        erase_config = ipw.Button(
+            description="Erase configuration", button_style="danger"
+        )
+        erase_config.on_click(self.erase_eln_configuration)
+
+        check_connection = ipw.Button(
+            description="Check connection", button_style="warning"
+        )
+        check_connection.on_click(self.check_connection)
+
+        self.display_eln_config()
+
+        super().__init__(
+            children=[
+                self.eln_instance,
+                self.eln_types,
+                self._output,
+                ipw.HBox([default_button, save_config, erase_config, check_connection]),
+            ],
+            **kwargs,
+        )
+
+    def update_list_of_elns(self):
+        config = self.get_configured_elns()
+        default_eln = config.pop("default", None)
+        self.eln_instance.options = [("Setup new ELN", {})] + [
+            (k, v) for k, v in config.items()
+        ]
+        if default_eln:
+            self.eln_instance.label = default_eln
+
+    def get_configured_elns(self):
+        try:
+            with open(ELN_CONFIG, "r") as file:
+                return json.load(file)
+        except (FileNotFoundError, json.JSONDecodeError, KeyError):
+            return {}
+
+    def set_current_eln_as_default(self, _=None):
+        self.update_eln_configuration("default", self.eln_instance.label)
+
+    def update_eln_configuration(self, eln, config):
+        elns = self.get_configured_elns()
+        elns[eln] = config
+        with open(ELN_CONFIG, "w") as file:
+            json.dump(elns, file, indent=4)
+
+    def erase_eln_configuration(self, _=None):
+        elns = self.get_configured_elns()
+        elns.pop(self.eln_instance.label, None)
+        if "default" in elns and elns["default"] not in elns:
+            elns.pop("default")
+
+        with open(ELN_CONFIG, "w") as file:
+            json.dump(elns, file, indent=4)
+        self.update_list_of_elns()
+
+    def check_connection(self, _=None):
+        print("Not implemented :(")
```

Comment:
Should this rather raise `NotImplementedError`?

Reply:
I've implemented a proper check.